packages/merge-styles/src/Stylesheet.ts

```diff
@@ -0,0 +1,161 @@
+import { GlobalSettings } from '@uifabric/utilities/lib/GlobalSettings';
+import { FabricPerformance } from '@uifabric/utilities/lib/FabricPerformance';
+import { IStyle } from './IStyle';
+const STYLESHEET_SETTING = 'stylesheet';
+
+export const enum InjectionMode {
+  none = 0,
+  insertNode = 1,
+  appendChild = 2
+}
+
+export interface IStyleSheetConfig {
+  async?: boolean;
+  injectionMode?: InjectionMode;
+}
+
+/**
+ *
+ * @public
+ */
+export class Stylesheet {
+  private _styleElement: HTMLStyleElement;
+  private _rules: string[];
+  private _config: IStyleSheetConfig;
+  private _rulesToInsert: string[];
+  private _timerId: number;
+  private _counter: number;
+  private _keyToClassName: { [key: string]: string };
+
+  // tslint:disable-next-line:no-any
+  private _classNameToArgs: { [key: string]: any };
+
+  public static getInstance(): Stylesheet {
```

Comment:
removed `GlobalSettings` usage. If window is unavailable i cache it in the local module.